cve_bin_tool/helper_script.py

```diff
@@ -0,0 +1,151 @@
+# Copyright (C) 2021 Intel Corporation
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+import os
+import re
+
+from cve_bin_tool.cvedb import CVEDB, DBNAME, DISK_LOCATION_DEFAULT
+from cve_bin_tool.extractor import Extractor
+from cve_bin_tool.log import LOGGER
+from cve_bin_tool.strings import Strings
+from cve_bin_tool.util import DirWalk, inpath
+from cve_bin_tool.version_scanner import (  # using it just for clean_file_path
+    VersionScanner,
+)
+
+
+class HelperScript:
+    """Helps Contributors to understand how various common filenames, version-strings, etc. look"""
+
+    dbname = os.path.join(DISK_LOCATION_DEFAULT, DBNAME)
+
+    def __init__(self, filename, vendor_name=None, product_name=None):
+        self.filename = filename
+        self.extractor = Extractor()
+        self.logger = LOGGER.getChild(self.__class__.__name__)
+        self.product = product_name or self.parse_filename(filename)
+        self.vendor = vendor_name
+
+        # for setting the data
+        self.connection = None
+        self.dbpath = os.path.join(DISK_LOCATION_DEFAULT, DBNAME)
+
+        # for extraction
+        self.walker = DirWalk().walk
+        self.file_stack = []
+
+        # for output
+        self.contain_patterns = []
+        self.filename_pattern = []
+        self.version_pattern = []
+        self.vendor_product = []
+
+    def extract_file(self):
+        with self.extractor as ectx:
+            if ectx.can_extract(self.filename):
+                binary_string_list = []
+                for filepath in self.walker([ectx.extract(self.filename)]):
+                    clean_path = VersionScanner.clean_file_path(
+                        filepath
+                    )  # to be removed???
+                    self.file_stack.append(f"{clean_path}")
+                    self.logger.info(f"scanning binaries from {clean_path}")
+                    # go to that path and scan for strings
+                    matches = self.search_pattern(
+                        self.parse_strings(filepath)
+                    )  # list of strings in binary
+                    binary_string_list.append(matches)
+                return binary_string_list
```

Comment:
The last meeting helped a lot with this!!! :)
The `clean_path` is for the logs and I think I would be removing it later. I'm parsing all the files in the .rpm file and then am searching for the desired strings in it and then returning these in a list.

I wanted suggestions over whether I should scan over all the files or not.*example*: busybox contains -
```
/data.cpio
/usr/lib/.build-id/1e/2a6ad3feaed3fbb308145c8db69f0dd936b427
/usr/share/man/man1/busybox.1.gz
/usr/share/doc/busybox/LICENSE
/usr/share/doc/busybox/README
/sbin/busybox
```

1) should I be scanning over `.cpio`?
2) The README and LICENSE should be ignored? The `man/` directory should be ignored?
3) Or should I just search in files which have the product_name in their filename (generally these files only contain the strings that we are looking for... this was the pattern that I observed) *or* should I just be ignoring files to scan?

Reply:
@peb-peb The scanfile routine in the VersionScanner class checks the type of file and unless it identifies as an executable binary, the file will be ignored. This will mean files such as LICENSE and README will autonatically be ignored. Try running the `file` command and see what it reports. Contents of an archive files (e.g. .gz) will be extracted to a temporary directory before checking the file.

Reply:
@anthonyharrison I've implemented these changes.